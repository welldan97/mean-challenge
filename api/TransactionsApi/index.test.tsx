/* eslint-disable @typescript-eslint/no-explicit-any */
import TransactionApi from '.';
import signatures from '../../lib/test/fixtures/signatures';
import solanaTransactions from '../../lib/test/fixtures/solanaTransactions';

const createPublicKeyMock = () => ({
  toBase58: jest.fn(() => 'publicKey'),
});

const createConnectionMock = () => ({
  getSignaturesForAddress: jest.fn(() => signatures),
  getParsedTransactions: jest.fn(() => solanaTransactions),
});

describe('TransactionsApi', () => {
  test('index', async () => {
    const publicKey = createPublicKeyMock();
    const connection = createConnectionMock();

    // Note: use real types
    const result = await TransactionApi.index(
      connection as any,
      publicKey as any,
    );

    // Note: better to use toEqual
    expect(result).toMatchSnapshot();
  });
});
