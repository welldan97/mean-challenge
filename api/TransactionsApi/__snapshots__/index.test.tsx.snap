// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionsApi getAll 1`] = `
Array [
  Object {
    "address": "F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23",
    "amount": 1000000000,
    "balanceAfter": 4940000000,
    "balanceBefore": 3940000000,
    "fee": 5000,
    "timestamp": 1665005686,
    "type": "receive",
    "walletAddress": "publicKey",
  },
  Object {
    "balanceAfter": 2538984320,
    "balanceBefore": 2549462160,
    "fee": 10000,
    "timestamp": 1665005634,
    "type": "other",
    "walletAddress": "publicKey",
  },
  Object {
    "address": "F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23",
    "amount": 1000000000,
    "balanceAfter": 3940000000,
    "balanceBefore": 2940000000,
    "fee": 5000,
    "timestamp": 1665005445,
    "type": "receive",
    "walletAddress": "publicKey",
  },
  Object {
    "address": "F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23",
    "amount": 40000000,
    "balanceAfter": 2940000000,
    "balanceBefore": 2900000000,
    "fee": 5000,
    "timestamp": 1664986617,
    "type": "receive",
    "walletAddress": "publicKey",
  },
  Object {
    "address": "F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23",
    "amount": 30000000,
    "balanceAfter": 2900000000,
    "balanceBefore": 2870000000,
    "fee": 5000,
    "timestamp": 1664982540,
    "type": "receive",
    "walletAddress": "publicKey",
  },
  Object {
    "address": "F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23",
    "amount": 30000000,
    "balanceAfter": 2870000000,
    "balanceBefore": 2840000000,
    "fee": 5000,
    "timestamp": 1664982449,
    "type": "receive",
    "walletAddress": "publicKey",
  },
  Object {
    "address": "F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23",
    "amount": 30000000,
    "balanceAfter": 2840000000,
    "balanceBefore": 2810000000,
    "fee": 5000,
    "timestamp": 1664977847,
    "type": "receive",
    "walletAddress": "publicKey",
  },
  Object {
    "address": "F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23",
    "amount": 30000000,
    "balanceAfter": 2810000000,
    "balanceBefore": 2780000000,
    "fee": 5000,
    "timestamp": 1664976815,
    "type": "receive",
    "walletAddress": "publicKey",
  },
  Object {
    "address": "F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23",
    "amount": 1000000000,
    "balanceAfter": 2780000000,
    "balanceBefore": 1780000000,
    "fee": 5000,
    "timestamp": 1664974742,
    "type": "receive",
    "walletAddress": "publicKey",
  },
  Object {
    "address": "9B5XszUGdMaxCZ7uSQhPzdks5ZQSmWxrmzCSvtJ6Ns6g",
    "amount": 2000000000,
    "balanceAfter": 4709497160,
    "balanceBefore": 2709497160,
    "fee": 5000,
    "timestamp": 1664928170,
    "type": "receive",
    "walletAddress": "publicKey",
  },
]
`;

exports[`TransactionsApi index 1`] = `
Array [
  Object {
    "address": "F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23",
    "amount": 1000000000,
    "balanceAfter": 4940000000,
    "balanceBefore": 3940000000,
    "fee": 5000,
    "timestamp": 1665005686,
    "type": "receive",
    "walletAddress": "publicKey",
  },
  Object {
    "balanceAfter": 2538984320,
    "balanceBefore": 2549462160,
    "fee": 10000,
    "timestamp": 1665005634,
    "type": "other",
    "walletAddress": "publicKey",
  },
  Object {
    "address": "F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23",
    "amount": 1000000000,
    "balanceAfter": 3940000000,
    "balanceBefore": 2940000000,
    "fee": 5000,
    "timestamp": 1665005445,
    "type": "receive",
    "walletAddress": "publicKey",
  },
  Object {
    "address": "F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23",
    "amount": 40000000,
    "balanceAfter": 2940000000,
    "balanceBefore": 2900000000,
    "fee": 5000,
    "timestamp": 1664986617,
    "type": "receive",
    "walletAddress": "publicKey",
  },
  Object {
    "address": "F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23",
    "amount": 30000000,
    "balanceAfter": 2900000000,
    "balanceBefore": 2870000000,
    "fee": 5000,
    "timestamp": 1664982540,
    "type": "receive",
    "walletAddress": "publicKey",
  },
  Object {
    "address": "F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23",
    "amount": 30000000,
    "balanceAfter": 2870000000,
    "balanceBefore": 2840000000,
    "fee": 5000,
    "timestamp": 1664982449,
    "type": "receive",
    "walletAddress": "publicKey",
  },
  Object {
    "address": "F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23",
    "amount": 30000000,
    "balanceAfter": 2840000000,
    "balanceBefore": 2810000000,
    "fee": 5000,
    "timestamp": 1664977847,
    "type": "receive",
    "walletAddress": "publicKey",
  },
  Object {
    "address": "F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23",
    "amount": 30000000,
    "balanceAfter": 2810000000,
    "balanceBefore": 2780000000,
    "fee": 5000,
    "timestamp": 1664976815,
    "type": "receive",
    "walletAddress": "publicKey",
  },
  Object {
    "address": "F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23",
    "amount": 1000000000,
    "balanceAfter": 2780000000,
    "balanceBefore": 1780000000,
    "fee": 5000,
    "timestamp": 1664974742,
    "type": "receive",
    "walletAddress": "publicKey",
  },
  Object {
    "address": "9B5XszUGdMaxCZ7uSQhPzdks5ZQSmWxrmzCSvtJ6Ns6g",
    "amount": 2000000000,
    "balanceAfter": 4709497160,
    "balanceBefore": 2709497160,
    "fee": 5000,
    "timestamp": 1664928170,
    "type": "receive",
    "walletAddress": "publicKey",
  },
]
`;

exports[`TransactionsApi index 2`] = `
Array [
  Object {
    "address": "F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23",
    "amount": 1000000000,
    "balanceAfter": 4940000000,
    "balanceBefore": 3940000000,
    "fee": 5000,
    "timestamp": 1665005686,
    "type": "receive",
    "walletAddress": "publicKey",
  },
  Object {
    "balanceAfter": 2538984320,
    "balanceBefore": 2549462160,
    "fee": 10000,
    "timestamp": 1665005634,
    "type": "other",
    "walletAddress": "publicKey",
  },
  Object {
    "address": "F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23",
    "amount": 1000000000,
    "balanceAfter": 3940000000,
    "balanceBefore": 2940000000,
    "fee": 5000,
    "timestamp": 1665005445,
    "type": "receive",
    "walletAddress": "publicKey",
  },
  Object {
    "address": "F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23",
    "amount": 40000000,
    "balanceAfter": 2940000000,
    "balanceBefore": 2900000000,
    "fee": 5000,
    "timestamp": 1664986617,
    "type": "receive",
    "walletAddress": "publicKey",
  },
  Object {
    "address": "F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23",
    "amount": 30000000,
    "balanceAfter": 2900000000,
    "balanceBefore": 2870000000,
    "fee": 5000,
    "timestamp": 1664982540,
    "type": "receive",
    "walletAddress": "publicKey",
  },
  Object {
    "address": "F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23",
    "amount": 30000000,
    "balanceAfter": 2870000000,
    "balanceBefore": 2840000000,
    "fee": 5000,
    "timestamp": 1664982449,
    "type": "receive",
    "walletAddress": "publicKey",
  },
  Object {
    "address": "F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23",
    "amount": 30000000,
    "balanceAfter": 2840000000,
    "balanceBefore": 2810000000,
    "fee": 5000,
    "timestamp": 1664977847,
    "type": "receive",
    "walletAddress": "publicKey",
  },
  Object {
    "address": "F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23",
    "amount": 30000000,
    "balanceAfter": 2810000000,
    "balanceBefore": 2780000000,
    "fee": 5000,
    "timestamp": 1664976815,
    "type": "receive",
    "walletAddress": "publicKey",
  },
  Object {
    "address": "F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23",
    "amount": 1000000000,
    "balanceAfter": 2780000000,
    "balanceBefore": 1780000000,
    "fee": 5000,
    "timestamp": 1664974742,
    "type": "receive",
    "walletAddress": "publicKey",
  },
  Object {
    "address": "9B5XszUGdMaxCZ7uSQhPzdks5ZQSmWxrmzCSvtJ6Ns6g",
    "amount": 2000000000,
    "balanceAfter": 4709497160,
    "balanceBefore": 2709497160,
    "fee": 5000,
    "timestamp": 1664928170,
    "type": "receive",
    "walletAddress": "publicKey",
  },
]
`;
